EngineLayer/ModernSearch/ModernSearchEngine.cs

```diff
@@ -191,38 +207,37 @@ protected List<int> GetBinsToSearch(Ms2ScanWithSpecificMass scan)
                 if (commonParameters.AddCompIons)
                 {
                     //okay, we're not actually adding in complementary m/z peaks, we're doing a shortcut and just straight up adding the bins assuming that they're z=1
-                    foreach (DissociationType dissociationType in DissociationTypes)
-                    {
-                        if (complementaryIonConversionDictionary.TryGetValue(dissociationType, out double protonMassShift))
-                        {
-                            protonMassShift = ClassExtensions.ToMass(protonMassShift, 1);
-                            int compFragmentFloorMass = (int)Math.Round(((scan.PrecursorMass + protonMassShift) * FragmentBinsPerDalton)) - obsFragmentCeilingMass;
-                            int compFragmentCeilingMass = (int)Math.Round(((scan.PrecursorMass + protonMassShift) * FragmentBinsPerDalton)) - obsFragmentFloorMass;
 
-                            // prevent index out of bounds errors
-                            if (compFragmentCeilingMass >= FragmentIndex.Length)
-                            {
-                                compFragmentCeilingMass = FragmentIndex.Length - 1;
+                    if (complementaryIonConversionDictionary.TryGetValue(commonParameters.DissociationType, out double protonMassShift)) //TODO: this is broken for EThcD because that method needs two conversions
```

Comment:
do we have unit tests that cover this code block?

Reply:
The non-specific tests should be covering this.